sdl_android/src/main/java/com/smartdevicelink/api/screen/TextAndGraphicManager.java

```diff
@@ -0,0 +1,890 @@
+			currentScreenData.setSecondaryGraphic(show.getSecondaryGraphic());
+		}
+	}
+
+	// Helpers
+
+	private List<String> findValidMainTextFields(){
+		List<String> array = new ArrayList<>();
+
+		if (textField1 != null && textField1.length() > 0) {
+			array.add(textField1);
+		}
+
+		if (textField2 != null && textField2.length() > 0) {
+			array.add(textField2);
+		}
+
+		if (textField3 != null && textField3.length() > 0) {
+			array.add(textField3);
+		}
+
+		if (textField4 != null && textField4.length() > 0) {
+			array.add(textField4);
+		}
+
+		return array;
+	}
+
+
+	private List<MetadataType> findNonNullMetadataFields(){
+		List<MetadataType> array = new ArrayList<>();
+
+		if (textField1Type != null) {
+			array.add(textField1Type);
+		}
+
+		if (textField2Type != null) {
+			array.add(textField2Type);
+		}
+
+		if (textField3Type != null) {
+			array.add(textField3Type);
+		}
+
+		if (textField4Type != null) {
+			array.add(textField4Type);
+		}
+
+		return array;
+	}
+
+	SdlArtwork getBlankArtwork(){
+
+		if (blankArtwork != null){
+			blankArtwork = new SdlArtwork();
+			blankArtwork.setType(FileType.GRAPHIC_PNG);
+			blankArtwork.setName("blankArtwork");
+			blankArtwork.setResourceId(R.drawable.transparent);
+		}
+		return blankArtwork;
+	}
+
+	private boolean isArtworkUploadedOrDoesntExist(SdlArtwork artwork){
+
+		if (fileManager.get() != null){
+			return artwork != null && fileManager.get().hasUploadedFile(artwork);
+		}
+
+		return false;
+	}
+
+	private boolean shouldUpdatePrimaryImage() {
+		if (displayCapabilities == null || displayCapabilities.getGraphicSupported()) {
+			if (currentScreenData.getGraphic() == null && primaryGraphic != null) {
+				return true;
+			} else if (currentScreenData.getGraphic() == null && primaryGraphic == null) {
+				return false;
+			}
+			return currentScreenData != null && (primaryGraphic != null && !currentScreenData.getGraphic().getValue().equalsIgnoreCase(primaryGraphic.getName()));
+		}
+		return false;
+	}
+
+	private boolean shouldUpdateSecondaryImage() {
+		// Cannot detect if there is a secondary image, so we'll just try to detect if there's a primary image and allow it if there is.
+		if (displayCapabilities == null || displayCapabilities.getGraphicSupported()) {
+			if (currentScreenData.getGraphic() == null && secondaryGraphic != null) {
+				return true;
+			} else if (currentScreenData.getGraphic() == null && secondaryGraphic == null) {
+				return false;
+			}
+			return currentScreenData != null && (secondaryGraphic != null && !currentScreenData.getGraphic().getValue().equalsIgnoreCase(secondaryGraphic.getName()));
+		}
+		return false;
+	}
+
+	int getNumberOfLines() {
+
+		if (displayCapabilities == null){
+			return 4;
+		}
+
+		int linesFound = 0;
+
+		List<TextField> textFields = displayCapabilities.getTextFields();
+
+		for (TextField field : textFields) {
+			if (field.getName() != null) {
+				TextFieldName name = field.getName();
```

Comment:
Move the declaration for `TexFieldName name;` before the `for` loop and just set the name through each iteration rather than create a new reference.